tst/gstreamer/gstkvstest.cpp

```diff
@@ -217,20 +292,47 @@ GST_START_TEST(check_playing_to_paused_and_back_to_playing)
         fail_unless_equals_int(gst_element_change_state(pElement, GST_STATE_CHANGE_PAUSED_TO_PLAYING), GST_STATE_CHANGE_SUCCESS);
         fail_unless_equals_int(gst_element_change_state(pElement, GST_STATE_CHANGE_PLAYING_TO_PAUSED), GST_STATE_CHANGE_SUCCESS);
         fail_unless_equals_int(gst_element_change_state(pElement, GST_STATE_CHANGE_PAUSED_TO_PLAYING), GST_STATE_CHANGE_SUCCESS);
+
+        gst_pad_set_active (srcpad, FALSE);
+
         cleanup_kinesisvideoproducersink(pElement);
     }
 GST_END_TEST;
 
+GST_START_TEST(test_check_credentials)
+    {
+        ck_abort_msg("Required environment variables (ACCESS_KEY and/or SECRET_KEY) are not set");
+    }
+GST_END_TEST;
+
 //Verify all State change events and direct state set events.
 
 Suite *gst_kinesisvideoproducer_suite(void) {
     Suite *s = suite_create("GstKinesisVideoSinkPlugin");
     TCase *tc = tcase_create("AllStateChangeTests");
-    accessKey = getenv(ACCESS_KEY_ENV_VAR);
-    secretKey = getenv(SECRET_KEY_ENV_VAR);
+    tcase_set_timeout(tc, 15);  // 15 second timeout per test
+
+    accessKey = GETENV(ACCESS_KEY_ENV_VAR);
+    secretKey = GETENV(SECRET_KEY_ENV_VAR);
+    sessionToken = GETENV(SESSION_TOKEN_ENV_VAR);
+
+    accessKey = accessKey ? accessKey : "";
+    secretKey = secretKey ? secretKey : "";
+    sessionToken = sessionToken ? sessionToken : "";
+
+    // Check if required environment variables are set
+    if (accessKey[0] == '\0' || secretKey[0] == '\0') {
```

Comment:
||sessionToken[0] == '\0'

Reply:
Not checking session token here since it can be empty string if permanent credentials are used. Added a comment.

We currently don't do further validation (e.g. regex validation) on the access key, secret key, session token variables in the rest of the SDK. We can look into adding that in the future.